### DISCUSSION TITLE
How to setup multiple instances in on premise installation?

### DISCUSSION THREAD

**Discussion by OndrejMus** [datalust/seq-tickets#2183] [Help] [answered]
Hello there,
I was looking for a way how to handle data segregation in situation when we want ingest from multiple solutions to one Seq server, that would not rely on content of logs/traces and filters. I have found following documentation: https://docs.datalust.co/v2/docs/named-instances but it is deprecated it I have not found similar description in up to date documentation.

Closest thing I have found in up to date documentation is -n parameter in sever commands (such as https://docs.datalust.co/docs/server-command-line#service-install or https://docs.datalust.co/docs/server-command-line#run), but it does not explain things like where data is stores, what url should used for ingestion etc.

Is there still an option to run multiple instances on same on premise installation (or similar feature)?

Thank you in advance :)

**Comment by nblumhardt** [comment, marked as answer]
Hi Ondrej! Thanks for your message.

If you need strict data segregation, or different sets of users accessing the logs, then multiple separate Seq instances are the way to go. For any significant production load, using either separate VMs (Windows) or container instances (Docker/Linux) is the recommended approach.

On Windows, the legacy "named instances" feature still functions; the `seq service install` command accepts a `--storage` argument which you can point to a separate folder for each instance. From there, commands like `seq config set -k api.listenUris` all accept a `-n <name>` argument that will cause the command to apply to the specified named instance.

If you _don't_ need strict data segregation or different sets of users, the better approach is to give each log source a unique Seq API key, and then in the configuration for that API key, specify a property such as `Solution = App1` which Seq will then apply server-side to all events ingested from that source.

Using unique API keys and tagging with properties on ingestion is much more efficient and easier to manage in the long run.

Let me know if this helps,
Nick

**Reply by OndrejMus** [reply]
Hello Nicolas,
Thank you for fast reply :) Having separate installation seems like overkill for our amount of logs. We might consider it in case customer would require strict segregation, but in that case, they would have to pay for VM and Seq license.

From what you are saying, I will check API key with custom property and predefined signals to filter logs and traces per solution.

Thank you for help.

Best regards,
Ondřej
